algorithms/src/std_algorithms/impl/Kokkos_RandomAccessIterator.hpp

```diff
@@ -73,16 +95,162 @@ class RandomAccessIterator< ::Kokkos::View<DataType, Args...> > {
+      : m_data(other.m_data), m_stride(other.m_stride) {}
 
   template <class OtherViewType,
             std::enable_if_t<std::is_convertible_v<OtherViewType, view_type>,
                              int> = 0>
   KOKKOS_FUNCTION RandomAccessIterator(
       const RandomAccessIterator<OtherViewType>& other)
-      : m_view(other.m_view), m_current_index(other.m_current_index) {}
+      : m_data(other.m_data), m_stride(other.m_stride) {}
 #endif
 
+  KOKKOS_FUNCTION
+  iterator_type& operator++() {
+    if constexpr (is_always_contiguous)
+      m_data++;
+    else
+      m_data += m_stride;
+    return *this;
+  }
+
+  KOKKOS_FUNCTION
+  iterator_type operator++(int) {
+    auto tmp = *this;
+    ++*this;
+    return tmp;
+  }
+
+  KOKKOS_FUNCTION
+  iterator_type& operator--() {
+    if constexpr (is_always_contiguous)
+      m_data--;
+    else
+      m_data -= m_stride;
+    return *this;
+  }
+
+  KOKKOS_FUNCTION
+  iterator_type operator--(int) {
+    auto tmp = *this;
+    --*this;
+    return tmp;
+  }
+
+  KOKKOS_FUNCTION
+  reference operator[](difference_type n) const {
+    if constexpr (is_always_contiguous)
+      return *(m_data + n);
+    else
+      return *(m_data + n * m_stride);
+  }
+
+  KOKKOS_FUNCTION
+  iterator_type& operator+=(difference_type n) {
+    if constexpr (is_always_contiguous)
+      m_data += n;
+    else
+      m_data += n * m_stride;
+    return *this;
+  }
+
+  KOKKOS_FUNCTION
+  iterator_type& operator-=(difference_type n) {
+    if constexpr (is_always_contiguous)
+      m_data -= n;
+    else
+      m_data -= n * m_stride;
+    return *this;
+  }
+
+  KOKKOS_FUNCTION
+  iterator_type operator+(difference_type n) const {
+    auto it = *this;
+    it += n;
+    return it;
+  }
+
+  KOKKOS_FUNCTION
+  iterator_type operator-(difference_type n) const {
+    auto it = *this;
+    it -= n;
+    return it;
+  }
+
+  KOKKOS_FUNCTION
+  difference_type operator-(iterator_type it) const {
+    if constexpr (is_always_contiguous)
+      return m_data - it.m_data;
+    else
+      return (m_data - it.m_data) / m_stride;
+  }
+
+  KOKKOS_FUNCTION
+  bool operator==(iterator_type other) const { return m_data == other.m_data; }
+
+  KOKKOS_FUNCTION
+  bool operator!=(iterator_type other) const { return m_data != other.m_data; }
+
+  KOKKOS_FUNCTION
+  bool operator<(iterator_type other) const { return m_data < other.m_data; }
+
+  KOKKOS_FUNCTION
+  bool operator<=(iterator_type other) const { return m_data <= other.m_data; }
+
+  KOKKOS_FUNCTION
+  bool operator>(iterator_type other) const { return m_data > other.m_data; }
+
+  KOKKOS_FUNCTION
+  bool operator>=(iterator_type other) const { return m_data >= other.m_data; }
+
+  KOKKOS_FUNCTION
+  reference operator*() const { return *m_data; }
+
+  KOKKOS_FUNCTION
+  pointer data() const { return m_data; }
+
+  KOKKOS_FUNCTION
+  int stride() const { return m_stride; }
+
+ private:
+  pointer m_data;
+  int m_stride;
```

Comment:
C++20 only but worth a thought:
```suggestion
  [[no_unique_address]] int m_stride;
```

Reply:
Please take a look at the following implementation of `std::span`. More specifically to how they deal with the extent that is compile-time or not. It might be a good source of inspiration for your compile-time or not stride.

https://github.com/gcc-mirror/gcc/blob/1de156eb2bb445cd0e0a582944dcd75d085f30c9/libstdc%2B%2B-v3/include/std/span#L103

This specilizaton of `RandomAccessIterator` really looks like a `std::span` with a *stride* instead of an *extent*.